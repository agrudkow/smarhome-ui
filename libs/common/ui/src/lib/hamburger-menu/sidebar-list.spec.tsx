import React from 'react';
import { render } from '@testing-library/react';

import SidebarList from './sidebar-list';

describe(' SidebarList', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<SidebarList />);
    expect(baseElement).toBeTruthy();
  });
});
