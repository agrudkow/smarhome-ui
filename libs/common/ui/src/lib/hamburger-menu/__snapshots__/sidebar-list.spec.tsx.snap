// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` SidebarList should match snapshot 1`] = `
.c1 {
  height: 45px;
  padding: 12px 16px;
  background-color: #4E4E4E;
  border-left: 2px solid #1BBDCE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: background-color 0.3s ease-in-out,border-left 0.3s ease-in-out;
  transition: background-color 0.3s ease-in-out,border-left 0.3s ease-in-out;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover {
  background-color: #4E4E4E;
}

.c3 {
  color: #555;
  font-size: 14px;
  line-height: 20px;
  -webkit-letter-spacing: -0.3px;
  -moz-letter-spacing: -0.3px;
  -ms-letter-spacing: -0.3px;
  letter-spacing: -0.3px;
  color: #1BBDCE;
  overflow: hidden;
  white-space: nowrap;
  padding-top: 2px;
  -webkit-transition: max-width 0.3s ease-in-out,opacity 0.3s ease-in-out, padding-left 0.3s ease-in-out,color 0.3s ease-in-out;
  transition: max-width 0.3s ease-in-out,opacity 0.3s ease-in-out, padding-left 0.3s ease-in-out,color 0.3s ease-in-out;
  max-width: 220px;
  opacity: 1;
  padding-left: 20px;
}

.c2 {
  min-width: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 path {
  -webkit-transition: fill 0.3s ease-in-out;
  transition: fill 0.3s ease-in-out;
  fill: #1BBDCE;
}

.c0 {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

@media (min-width:769px) {
  .c3 {
    font-size: 14px;
    line-height: 20px;
  }
}

@media (min-width:1024px) {
  .c3 {
    font-size: 16px;
    line-height: 24px;
  }
}

@media (min-width:1024px) {
  .c3 {
    font-size: 20px;
    line-height: 26px;
  }
}

<div>
  <ul
    class="c0"
  >
    <li>
      <a
        aria-current="page"
        class="c1 active"
        data-testid="styled-sidebar-list-item"
        href="/"
      >
        <div
          class="c2"
        >
          <div />
        </div>
        <div
          class="c3"
          data-testid="text"
        />
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        class="c1 active"
        data-testid="styled-sidebar-list-item"
        href="/"
      >
        <div
          class="c2"
        >
          <div />
        </div>
        <div
          class="c3"
          data-testid="text"
        />
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        class="c1 active"
        data-testid="styled-sidebar-list-item"
        href="/"
      >
        <div
          class="c2"
        >
          <div />
        </div>
        <div
          class="c3"
          data-testid="text"
        />
      </a>
    </li>
  </ul>
</div>
`;
